### PR TITLE
Deploy our OCI artifact to a private registry

### PR DESCRIPTION
**Pull request by chouquette**
### What does this PR do?

This PR starts deploying our OCI artifacts to an internal registry.

### Additional Notes

This is probably a bit too `datadog-agent` centric and should be reworked to have a base job that handles multiple flavors, but we'll see about that when we get there

### CODE REVIEW

**Review by Pythyu — APPROVED** [review]

File: .gitlab/deploy_packages/deploy_common.yml [hunk 1]
```diff
@@ -79,12 +79,22 @@ deploy_packages_oci:
...
+    - set +x
+    - export GITHUB_KEY_B64=$($CI_PROJECT_DIR/tools/ci/aws_ssm_get_wrapper.sh ci.datadog-agent.platform-github-app-key)
+    - export GITHUB_APP_ID=682216
+    - export GH_TOKEN=$(inv -e github.get-token-from-app --app-id-env=GITHUB_APP_ID --pkey-env=GITHUB_KEY_B64)
+    - export VERSION=$(inv agent.version -u)
+    - git config --global credential.helper '!f() { echo username=x-access-token; echo "password=$GH_TOKEN"; };f'
+    - git clone https://github.com/DataDog/datadog-packages /tmp/datadog-packages
+    - cd /tmp/datadog-packages/cmd/datadog-package
+    - go build .
+    - ./datadog-package push registry.ddbuild.io/ci/datadog-agent/datadog-agent-ociru:${VERSION} ${OMNIBUS_PACKAGE_DIR}/datadog-agent-${MAJOR_VERSION}.*.oci.tar
```
Comment (BaptisteFoy) [inline .gitlab/deploy_packages/deploy_common.yml:97]: **💬 Suggestion:** ‏Can we use `registry.ddbuild.io/ci/remote-updates/datadog-agent:${VERSION}`? That'd match the tracers

Comment (chouquette) [reply .gitlab/deploy_packages/deploy_common.yml:97]: For `registry.ddbuild.io/ci/remote-updates/` no problem at all on my end.

Regarding the image name itself, I believe @vboulineau wanted to have a more descriptive name than `datadog-agent`.
The prefix/folder might be enough to disambiguate what kind of `datadog-package` though

**Review by BaptisteFoy — APPROVED** [review]

File: .gitlab/deploy_packages/deploy_common.yml (hunk 1 above)
Comment (BaptisteFoy) [reply .gitlab/deploy_packages/deploy_common.yml]: True, the name may change again, but it's easy to change.

**Review by amenasria — APPROVED** [review]
LGTM for Agent CI Exp owned file !

**Pull request merged**